addons/mail/static/src/messaging/entity/activity/activity.js

```diff
@@ -0,0 +1,200 @@
+                kwargs: { feedback },
+            });
+            this.chatter.refresh();
+            this.delete();
+            return action;
+        }
+
+        //----------------------------------------------------------------------
+        // Private
+        //----------------------------------------------------------------------
+
+        /**
+         * @override
+         */
+        _update(data) {
+            let {
+                activity_category: category,
+                activity_type_id: [
+                    activityTypeId,
+                    activityTypeDisplayName,
+                ] = [],
+                can_write: canWrite,
+                create_date: dateCreate,
+                create_uid: [
+                    creatorId,
+                    creatorDisplayName,
+                ] = [],
+                date_deadline: dateDeadline,
+                forceNext = this.forceNext || false,
+                icon,
+                id = this.id,
+                mail_template_ids = [],
+                note,
+                res_id = this.res_id,
+                res_model: res_model = this.res_model,
+                state,
+                summary,
+                user_id: [
+                    assigneeId,
+                    assigneeDisplayName,
+                ] = [],
+            } = data;
+
+            this._write({
+                canWrite,
+                category,
+                dateCreate,
+                dateDeadline,
+                forceNext,
+                icon,
+                id,
+                note,
+                res_id,
+                res_model,
+                state,
+                summary,
+            });
+
+            // activity_type
+            if (activityTypeId) {
+                const type = this.env.entities.ActivityType.insert({
+                    displayName: activityTypeDisplayName,
+                    id: activityTypeId,
+                });
+                this.link({ type });
+            }
+            // assignee
+            if (assigneeId) {
+                const assignee = this.env.entities.User.insert({
+                    displayName: assigneeDisplayName,
+                    id: assigneeId,
+                });
+                this.link({ assignee });
+            }
+            // creator
+            if (creatorId) {
+                const creator = this.env.entities.User.insert({
+                    displayName: creatorDisplayName,
+                    id: creatorId,
+                });
+                this.link({ creator });
+            }
+            // mail templates
+            for (const mailTemplateData of mail_template_ids) {
+                const mailTemplate = this.env.entities.MailTemplate.insert(mailTemplateData);
+                this.link({ mailTemplates: mailTemplate });
+            }
+        }
+
+    }
+
+    Object.assign(Activity, {
+        relations: Object.assign({}, Entity.relations, {
```

Comment:
It's annoying that we always have to use this syntax to get the relations from the parent.